spring-data-cassandra/src/main/java/org/springframework/data/cassandra/repository/support/SimpleReactiveCassandraRepository.java

```diff
@@ -304,6 +305,14 @@ public SimpleReactiveCassandraRepository(CassandraEntityInformation<T, ID> metad
 		return Flux.fromIterable(entities).flatMap(this.operations::delete).then();
 	}
 
+	@Override
+	public Mono<Void> deleteAllById(Iterable<? extends ID> ids) {
+
+		Assert.notNull(ids, "The given Iterable of entities must not be null");
+
+		return Flux.fromIterable(ids).flatMap(this::deleteById).then();
```

Comment:
We should use the same semantics as with `findAllById(…)` regarding the query mechanism to avoid individual database queries for each ID.